### PR TITLE
HIVE-25152: Remove Superfluous Logging Code

### PR DESCRIPTION
**Pull request by belugabehr**

### CODE REVIEW

**Review by miklosgergely — APPROVED** [review]

File: ql/src/java/org/apache/hadoop/hive/ql/exec/vector/reducesink/VectorReduceSinkCommonOperator.java [hunk 1]
```diff
@@ -267,9 +267,7 @@ protected void initializeOp(Configuration hconf) throws HiveException {
     reduceSkipTag = conf.getSkipTag();
     reduceTagByte = (byte) conf.getTag();
 
-    if (LOG.isInfoEnabled()) {
-      LOG.info("Using tag = " + (int) reduceTagByte);
-    }
+    LOG.info("Using tag = " + reduceTagByte);
```
Comment (pgaref) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/vector/reducesink/VectorReduceSinkCommonOperator.java:270]: Using tag = {}

File: ql/src/java/org/apache/hadoop/hive/ql/exec/vector/reducesink/VectorReduceSinkCommonOperator.java [hunk 2]
```diff
@@ -382,9 +380,7 @@ protected void closeOp(boolean abort) throws HiveException {
     super.closeOp(abort);
     out = null;
     reducerHash = null;
-    if (LOG.isInfoEnabled()) {
-      LOG.info(toString() + ": records written - " + numRows);
-    }
+    LOG.info(this + ": records written - " + numRows);
```
Comment (pgaref) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/vector/reducesink/VectorReduceSinkCommonOperator.java:383]: {} : records written - {}

File: ql/src/java/org/apache/hadoop/hive/ql/io/HiveInputFormat.java [hunk 3]
```diff
@@ -313,17 +307,13 @@ public void configure(JobConf job) {
...
+    LOG.debug("Checking {} against {}", ifName, formatList);
     String[] formats = StringUtils.getStrings(formatList);
     if (formats != null) {
       for (String format : formats) {
         // TODO: should we check isAssignableFrom?
         if (ifName.equals(format)) {
-          if (LOG.isInfoEnabled()) {
-            LOG.info("Using SerDe-based LLAP reader for " + ifName);
-          }
+          LOG.info("Using SerDe-based LLAP reader for " + ifName);
```
Comment (pgaref) [inline ql/src/java/org/apache/hadoop/hive/ql/io/HiveInputFormat.java:316]: Using SerDe-based LLAP reader for {}

File: ql/src/java/org/apache/hadoop/hive/ql/io/HiveInputFormat.java [hunk 4]
```diff
@@ -263,22 +263,16 @@ public void configure(JobConf job) {
       isSupported = isSerdeBased = checkInputFormatForLlapEncode(conf, ifName);
     }
     if ((!isSupported || !isVectorized) && !isCacheOnly) {
-      if (LOG.isInfoEnabled()) {
-        LOG.info("Not using llap for " + ifName + ": supported = "
-          + isSupported + ", vectorized = " + isVectorized + ", cache only = " + isCacheOnly);
-      }
+      LOG.info("Not using llap for " + ifName + ": supported = " + isSupported + ", vectorized = " + isVectorized
```
Comment (pgaref) [inline ql/src/java/org/apache/hadoop/hive/ql/io/HiveInputFormat.java:266]: same

File: ql/src/java/org/apache/hadoop/hive/ql/io/HiveInputFormat.java [hunk 5]
```diff
@@ -831,9 +817,7 @@ private static void processForWriteIdsForMmRead(Path dir, Configuration conf,
     DeltaFilesMetricReporter.backPropagateAcidMetrics(job, newjob);
 
     Utilities.clearWorkMapForConf(job);
-    if (LOG.isInfoEnabled()) {
-      LOG.info("number of splits " + result.size());
-    }
+    LOG.info("number of splits " + result.size());
```
Comment (pgaref) [inline ql/src/java/org/apache/hadoop/hive/ql/io/HiveInputFormat.java:820]: number of splits {}

File: ql/src/java/org/apache/hadoop/hive/ql/optimizer/ConstantPropagateProcFactory.java [hunk 6]
```diff
@@ -1463,15 +1441,11 @@ public Object process(Node nd, Stack<Node> stack, NodeProcessorCtx ctx, Object..
...
-            LOG.debug("Skip JOIN-FIL(*)-RS structure.");
-          }
+          LOG.debug("Skip JOIN-FIL(*)-RS structure.");
           return null;
         }
       }
-      if (LOG.isInfoEnabled()) {
-        LOG.info("Old exprs " + conf.getExprs());
-      }
+      LOG.info("Old exprs " + conf.getExprs());
```
Comment (pgaref) [inline ql/src/java/org/apache/hadoop/hive/ql/optimizer/ConstantPropagateProcFactory.java:1448]: Old exprs {}

File: ql/src/java/org/apache/hadoop/hive/ql/optimizer/ConstantPropagateProcFactory.java [hunk 7]
```diff
@@ -1484,18 +1458,14 @@ public Object process(Node nd, Stack<Node> stack, NodeProcessorCtx ctx, Object..
         for (ExprNodeDesc expr : exprs) {
           ExprNodeDesc newExpr = foldExpr(expr, constants, cppCtx, op, tag, false);
           if (newExpr instanceof ExprNodeConstantDesc) {
-            if (LOG.isInfoEnabled()) {
-              LOG.info("expr " + newExpr + " fold from " + expr + " is removed.");
-            }
+            LOG.info("expr " + newExpr + " fold from " + expr + " is removed.");
```
Comment (pgaref) [inline ql/src/java/org/apache/hadoop/hive/ql/optimizer/ConstantPropagateProcFactory.java:1461]: same

File: ql/src/java/org/apache/hadoop/hive/ql/optimizer/ConstantPropagateProcFactory.java [hunk 8]
```diff
@@ -1484,18 +1458,14 @@ public Object process(Node nd, Stack<Node> stack, NodeProcessorCtx ctx, Object..
...
             continue;
           }
           newExprs.add(newExpr);
         }
         e.setValue(newExprs);
       }
-      if (LOG.isInfoEnabled()) {
-        LOG.info("New exprs " + conf.getExprs());
-      }
+      LOG.info("New exprs " + conf.getExprs());
```
Comment (pgaref) [inline ql/src/java/org/apache/hadoop/hive/ql/optimizer/ConstantPropagateProcFactory.java:1468]: same

File: ql/src/java/org/apache/hadoop/hive/ql/optimizer/RedundantDynamicPruningConditionsRemoval.java [hunk 9]
```diff
@@ -155,9 +155,7 @@ public Object process(Node nd, Stack<Node> stack, NodeProcessorCtx procCtx, Obje
                 break;
               }
             }
-            if (LOG.isInfoEnabled()) {
-              LOG.info("Dynamic pruning condition removed: " + child);
-            }
+            LOG.info("Dynamic pruning condition removed: " + child);
```
Comment (pgaref) [inline ql/src/java/org/apache/hadoop/hive/ql/optimizer/RedundantDynamicPruningConditionsRemoval.java:158]: Dynamic pruning condition removed: {}

File: ql/src/java/org/apache/hadoop/hive/ql/optimizer/physical/MemoryDecider.java [hunk 10]
```diff
@@ -197,9 +197,7 @@ public int compare(MapJoinOperator mj1, MapJoinOperator mj2) {
           }
 
           if (size < remainingSize) {
-            if (LOG.isInfoEnabled()) {
-              LOG.info("Setting " + size + " bytes needed for " + mj + " (in-mem)");
-            }
+            LOG.info("Setting " + size + " bytes needed for " + mj + " (in-mem)");
```
Comment (pgaref) [inline ql/src/java/org/apache/hadoop/hive/ql/optimizer/physical/MemoryDecider.java:200]: same

File: ql/src/java/org/apache/hadoop/hive/ql/optimizer/physical/MemoryDecider.java [hunk 11]
```diff
@@ -226,23 +224,17 @@ public int compare(MapJoinOperator mj1, MapJoinOperator mj2) {
 
         for (MapJoinOperator mj : sortedMapJoins) {
           long size = (long)(weight * sizes.get(mj));
-          if (LOG.isInfoEnabled()) {
-            LOG.info("Setting " + size + " bytes needed for " + mj + " (spills)");
-          }
+          LOG.info("Setting " + size + " bytes needed for " + mj + " (spills)");
```
Comment (pgaref) [inline ql/src/java/org/apache/hadoop/hive/ql/optimizer/physical/MemoryDecider.java:227]: same

File: ql/src/java/org/apache/hadoop/hive/ql/optimizer/physical/MemoryDecider.java [hunk 12]
```diff
@@ -226,23 +224,17 @@ public int compare(MapJoinOperator mj1, MapJoinOperator mj2) {
...
-        if (LOG.isInfoEnabled()) {
-          LOG.info("Scaling mapjoin memory w/o stats");
-        }
+        LOG.info("Scaling mapjoin memory w/o stats");
 
         for (MapJoinOperator mj : mapJoins) {
-          if (LOG.isInfoEnabled()) {
-            LOG.info("Setting " + size + " bytes needed for " + mj + " (fallback)");
-          }
+          LOG.info("Setting " + size + " bytes needed for " + mj + " (fallback)");
```
Comment (pgaref) [inline ql/src/java/org/apache/hadoop/hive/ql/optimizer/physical/MemoryDecider.java:237]: same

File: accumulo-handler/src/java/org/apache/hadoop/hive/accumulo/predicate/AccumuloPredicateHandler.java [hunk 13]
```diff
@@ -320,9 +320,7 @@ protected Object generateRanges(Configuration conf, ColumnMapper columnMapper,
         itrs.add(toSetting(mapping, sc, binaryEncodedRow));
       }
     }
-    if (LOG.isInfoEnabled()) {
-      LOG.info("num iterators = " + itrs.size());
-    }
+    LOG.info("num iterators = " + itrs.size());
```
Comment (pgaref) [inline accumulo-handler/src/java/org/apache/hadoop/hive/accumulo/predicate/AccumuloPredicateHandler.java:323]: num iterators = {}

File: llap-client/src/java/org/apache/hadoop/hive/llap/registry/impl/SlotZnode.java [hunk 14]
```diff
@@ -256,9 +256,7 @@ private void processWatchedEvent(WatchedEvent event) {
   private void processCreateResult(CuratorFramework client, CuratorEvent event) throws Exception {
     boolean doesExist = event.getResultCode() == KeeperException.Code.NODEEXISTS.intValue();
     if (!doesExist && event.getResultCode() != KeeperException.Code.OK.intValue()) {
-      if (LOG.isInfoEnabled()) {
-        LOG.info("Trying to reacquire due to create error: " + event);
-      }
+      LOG.info("Trying to reacquire due to create error: " + event);
```
Comment (pgaref) [inline llap-client/src/java/org/apache/hadoop/hive/llap/registry/impl/SlotZnode.java:259]: Trying to reacquire due to create error: {}

File: llap-server/src/java/org/apache/hadoop/hive/llap/security/LlapServerSecurityInfo.java [hunk 15]
```diff
@@ -33,9 +33,7 @@
 
   @Override
   public KerberosInfo getKerberosInfo(Class<?> protocol, Configuration conf) {
-    if (LOG.isDebugEnabled()) {
-      LOG.debug("Trying to get KerberosInfo for " + protocol);
-    }
+    LOG.debug("Trying to get KerberosInfo for " + protocol);
```
Comment (pgaref) [inline llap-server/src/java/org/apache/hadoop/hive/llap/security/LlapServerSecurityInfo.java:36]: Trying to reacquire due to create error: {}

File: llap-tez/src/java/org/apache/hadoop/hive/llap/tezplugins/LlapTaskCommunicator.java [hunk 16]
```diff
@@ -164,9 +164,7 @@ public LlapTaskCommunicator(
     } else {
       this.token = null;
     }
-    if (LOG.isInfoEnabled()) {
-      LOG.info("Task communicator with a token " + token);
-    }
+    LOG.info("Task communicator with a token " + token);
```
Comment (pgaref) [inline llap-tez/src/java/org/apache/hadoop/hive/llap/tezplugins/LlapTaskCommunicator.java:167]: Task communicator with a token {}

File: ql/src/java/org/apache/hadoop/hive/ql/exec/AppMasterEventOperator.java [hunk 17]
```diff
@@ -100,19 +100,16 @@ public void process(Object row, int tag) throws HiveException {
       Writable writableRow = serializer.serialize(row, rowInspector);
       writableRow.write(buffer);
       if (buffer.getLength() > MAX_SIZE) {
-        if (LOG.isInfoEnabled()) {
-          LOG.info("Disabling AM events. Buffer size too large: " + buffer.getLength());
-        }
+        LOG.info("Disabling AM events. Buffer size too large: " + buffer.getLength());
```
Comment (pgaref) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/AppMasterEventOperator.java:103]: Disabling AM events. Buffer size too large: {}

File: ql/src/java/org/apache/hadoop/hive/ql/exec/AppMasterEventOperator.java [hunk 18]
```diff
@@ -137,10 +134,8 @@ public void closeOp(boolean abort) throws HiveException {
           InputInitializerEvent.create(vertexName, inputName,
               ByteBuffer.wrap(payload, 0, payload.length));
 
-      if (LOG.isInfoEnabled()) {
-        LOG.info("Sending Tez event to vertex = " + vertexName + ", input = " + inputName
-          + ". Payload size = " + payload.length);
-      }
+      LOG.info("Sending Tez event to vertex = " + vertexName + ", input = " + inputName + ". Payload size = "
```
Comment (pgaref) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/AppMasterEventOperator.java:137]: same

File: ql/src/java/org/apache/hadoop/hive/ql/exec/CommonJoinOperator.java [hunk 19]
```diff
@@ -391,9 +391,7 @@ protected void initializeOp(Configuration hconf) throws HiveException {
       }
     }
 
-    if (LOG.isInfoEnabled()) {
-      LOG.info("JOIN " + outputObjInspector.getTypeName() + " totalsz = " + totalSz);
-    }
+    LOG.info("JOIN " + outputObjInspector.getTypeName() + " totalsz = " + totalSz);
```
Comment (pgaref) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/CommonJoinOperator.java:394]: same

File: ql/src/java/org/apache/hadoop/hive/ql/exec/DemuxOperator.java [hunk 20]
```diff
@@ -185,10 +185,7 @@ protected void initializeOp(Configuration hconf) throws HiveException {
       }
       newChildOperatorsTag[i] = toArray(childOperatorTags);
     }
-    if (LOG.isInfoEnabled()) {
-      LOG.info("newChildOperatorsTag " + Arrays.toString(newChildOperatorsTag));
-    }
-
+    LOG.info("newChildOperatorsTag " + Arrays.toString(newChildOperatorsTag));
```
Comment (pgaref) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/DemuxOperator.java:188]: same

File: ql/src/java/org/apache/hadoop/hive/ql/exec/DemuxOperator.java [hunk 21]
```diff
@@ -211,15 +208,11 @@ protected void initializeOp(Configuration hconf) throws HiveException {
   @Override
   protected void initializeChildren(Configuration hconf) throws HiveException {
     state = State.INIT;
-    if (LOG.isInfoEnabled()) {
-      LOG.info("Operator " + id + " " + getName() + " initialized");
-      LOG.info("Initializing children of " + id + " " + getName());
-    }
+    LOG.info("Operator " + id + " " + getName() + " initialized");
```
Comment (pgaref) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/DemuxOperator.java:211]: All below need to switch to {}

File: ql/src/java/org/apache/hadoop/hive/ql/exec/FileSinkOperator.java [hunk 22]
```diff
@@ -422,11 +420,9 @@ public void initializeBucketPaths(int filesIdx, String taskId, boolean isNativeT
           finalPaths[filesIdx] = finalPath;
           outPaths[filesIdx] = finalPath;
         }
-        if (LOG.isInfoEnabled()) {
-          LOG.info("Final Path: FS " + finalPaths[filesIdx]);
-          if (LOG.isInfoEnabled() && (!isMmTable && !isDirectInsert)) {
-            LOG.info("Writing to temp file: FS " + outPaths[filesIdx]);
-          }
+        LOG.info("Final Path: FS " + finalPaths[filesIdx]);
```
Comment (pgaref) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/FileSinkOperator.java:423]: {}

File: ql/src/java/org/apache/hadoop/hive/ql/exec/MapJoinOperator.java [hunk 23]
```diff
@@ -831,9 +827,7 @@ public void closeOp(boolean abort) throws HiveException {
         }
       }
 
-      if (LOG.isInfoEnabled()) {
-        LOG.info("spilled: " + spilled + " abort: " + abort + ". Clearing spilled partitions.");
-      }
+      LOG.info("spilled: " + spilled + " abort: " + abort + ". Clearing spilled partitions.");
```
Comment (pgaref) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/MapJoinOperator.java:830]: same

File: ql/src/java/org/apache/hadoop/hive/ql/exec/FileSinkOperator.java [hunk 24]
```diff
@@ -896,9 +886,7 @@ protected void createBucketForFileIdx(FSPaths fsp, int filesIdx)
           + "; out path " + fsp.outPaths[filesIdx] +" (spec path " + specPath + ", tmp path "
           + fsp.buildTmpPath() + ", task " + taskId + ")");
       }
-      if (LOG.isInfoEnabled()) {
-        LOG.info("New Final Path: FS " + fsp.finalPaths[filesIdx]);
-      }
+      LOG.info("New Final Path: FS " + fsp.finalPaths[filesIdx]);
```
Comment (pgaref) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/FileSinkOperator.java:889]: same

**Comment by belugabehr** [comment]
@pgaref Thank you so much for your interest in this.

I do not believe the requested changes will yield any value.  The two reasons one would use the anchors `{}` in SLF4J is twofold:

1. Readability (String formatting and no need for boilerplate logging guards)
2. Performance when _not_ logging

This PR is simply to remove superfluous code that does not add any measurable value and makes the code harder to read.  I don't want to be adding in new code changes if I can help it.

With that said, in any production environment, the logging level will always be _at least_: INFO, WARN, ERROR.  Therefore, there is no need to ever care about performance of INFO messages.  It is _assumed_ that they will always be printed, so there is no need to add logging guards (`if (Log.isInfoEnabled())`) and there is no need for anchors.  Again, anchors are nice for readability, but I don't want to modify the existing INFO level logging too much; that is another exercise for another PR if someone is truly inclined.  I think it's better to just stop introducing this superfluous code moving forward and the old stuff will eventually be updated.

Thanks.

**Comment by belugabehr** [comment]
... continued...

That is to say:

```java
LOG.info("New Final Path: FS " + fsp.finalPaths[filesIdx]);
LOG.info("New Final Path: FS {}", fsp.finalPaths[filesIdx]);
```

These two statements, will always produce the same output (as INFO is on be default in every production under the sun).  However, the second one will always have the overhead of having to find the anchor in the format string and then replace it with the value.  The first example, there is simply a string concatenation that occurs.  This will be faster and I don't find this log particularly hard to read.

**Comment by pgaref** [comment]
> ... continued...
> 
> That is to say:
> 
> ```java
> LOG.info("New Final Path: FS " + fsp.finalPaths[filesIdx]);
> LOG.info("New Final Path: FS {}", fsp.finalPaths[filesIdx]);
> ```
> 
> These two statements, will always produce the same output (as INFO is on by default in every production environment under the sun). However, the second one will always have the overhead of having to find the anchor in the format string and then replace it with the value. The first example, there is simply a string concatenation that occurs. This will be faster and I don't find this log particularly hard to read.

Hey @belugabehr thanks for the details! I agree adding anchors on all INFO statements wont add and perf value here considering that INFO is the default log lvl -- I was mostly thinking consistency when posting the comments, eventually using a single logging style but maybe thats out of the scope of this PR.

Happy to +1 as is and maybe discuss this as part of a new ticket

**Pull request merged**